bump deps

0z13 opened the pull request:
The biggest change is adding a context to sig-storage-lib-external-provisioner libs, also added the patches from our deployment repo.

0z13 committed:
bump io/sig-storage-lib-external-provisioner

0z13 committed:
chore: accommodate API changes

0z13 committed:
go mod tidy

0z13 committed:
.gitignore

0z13 committed:
remove Dockerfile, Makefile

0z13 committed:
apply 0001-secret-localfs.patch

0z13 committed:
apply 00000-to-python3 patch

0z13 committed:
add flake.nix

0z13 committed:
gofmt

srhb commented on cephfs-provisioner.go:
```diff
@@ -425,7 +421,7 @@ func main() {
 	// The controller needs to know what the server version is because out-of-tree
 	// provisioners aren't officially supported until 1.5
-	serverVersion, err := clientset.Discovery().ServerVersion()
+	// serverVersion, err := clientset.Discovery().ServerVersion()
```
Drop this and above comments while we can remember it's not needed?

0z13 replied:
oh yup, removed

srhb reviewed (changes requested):
Looks great. Please add an overlay to the flake for consumption downstream. It would also be nice if `nix build` worked out of the box.

0z13 committed:
nixpkgs: Pin at 22.11

To make this build.

python needs to be bumped to python310 when nixpkgs.follows in deployments gets bumped to 23.05

0z13 committed:
flake.nix: add overlay

srhb commented on flake.nix:
```diff
@@ -0,0 +1,77 @@
+    flake-parts.lib.mkFlake {inherit inputs;} 
+
+    ({ withSystem, ... }: {
+      flake.overlays.default = final: prev:
```
I'm not sure what's going on here. The overlay should look like `final: prev: { cephfs-provisioner = final.buildGoModule [...];}` (or something like that) and you should consume it in your package definition instead. Currently neither final nor prev appears anywhere on the RHS of =, so the package set used to compose overlays is never used.

0z13 replied:
Thanks for helping me through this sticking point, should be good now :)

0z13 committed:
drop old comment around unsupported api
